src/server/status/metrics.js

```diff
@@ -1,39 +1,67 @@
-import _ from 'lodash';
+import { get, set, isObject } from 'lodash';
 import { keysToSnakeCaseShallow } from '../../utils/case_conversion';
+import { getAllStats as cGroupStats } from './cgroup';
+
+let cGroupStatsAvailable = true;
 
 export function collectMetrics(kbnServer, server, config) {
   server.plugins['even-better'].monitor.on('ops', function (event) {
-    kbnServer.metrics = getMetrics({ event, config });
+    getMetrics({ event, config }).then(data => { kbnServer.metrics = data; });
   });
 }
 
-export function getMetrics({ event, config }) {
+export async function getMetrics({ event, config }) {
   const port = config.get('server.port');
   const timestamp = new Date().toISOString();
-  return {
+  const cgroup = await cGroupStatsIfAvailable();
+
+  const metrics = {
     last_updated: timestamp,
     collection_interval_in_millis: config.get('ops.interval'),
     uptime_in_millis: process.uptime() * 1000,
     process: {
       mem: {
-        heap_max_in_bytes: _.get(event, 'psmem.heapTotal'),
-        heap_used_in_bytes:  _.get(event, 'psmem.heapUsed')
+        heap_max_in_bytes: get(event, 'psmem.heapTotal'),
+        heap_used_in_bytes:  get(event, 'psmem.heapUsed')
       }
     },
     os: {
       cpu: {
         load_average: {
-          '1m': _.get(event, 'osload.0'),
-          '5m': _.get(event, 'osload.1'),
-          '15m': _.get(event, 'osload.1')
+          '1m': get(event, 'osload.0'),
+          '5m': get(event, 'osload.1'),
+          '15m': get(event, 'osload.1')
         }
       }
     },
     response_times: {
-      avg_in_millis:  _.get(event, ['responseTimes', port, 'avg']),
-      max_in_millis: _.get(event, ['responseTimes', port, 'max'])
+      avg_in_millis:  get(event, ['responseTimes', port, 'avg']),
+      max_in_millis: get(event, ['responseTimes', port, 'max'])
     },
-    requests:  keysToSnakeCaseShallow(_.get(event, ['requests', port])),
-    concurrent_connections: _.get(event, ['concurrents', port])
+    requests:  keysToSnakeCaseShallow(get(event, ['requests', port])),
+    concurrent_connections: get(event, ['concurrents', port])
   };
+
+  async function cGroupStatsIfAvailable() {
+    if (!cGroupStatsAvailable) {
+      return;
+    }
+
+    const cgroup = await cGroupStats({
+      cpuPath: config.get('cpu.cgroup.path.override') ? '/' : undefined,
```

Comment:
👍  for consistency.  it's not clear to me from the thread if override is meant to be a bool, the current (eventually deprecated) implementation in es looks like a string.  if i'm missing something feel free to ignore.